### ISSUE TITLE
Mismatched 0-indexing and 1-indexing in Solution._single_frame()

### ISSUE THREAD

**Issue by orioncohen** [MDAnalysis/solvation-analysis#45]
A colleague encountered a bug caused my mismatched 0-indexing vs 1-indexing. Python generally 0-indexes but MDAnalysis atom id's are 1-indexed. In the solution class, `self.u.atoms` is fancy indexed by a column of `pairs_array`, which contains atom ids. This causes all atoms to be shifted by 1, and if the pairs_arrays contains the final atom in the universe, it causes an indexing error. 

I am still exploring the implications of this bug, but the fix is relatively straightforward. Fixing it may cause some of the tests to fail if it changes any values.

```python
        pairs_array = np.concatenate(pairs_list, dtype=int)
        dist_array = np.concatenate(dist_list)
        res_name_array = np.concatenate(tags_list)
        res_id_array = self.u.atoms[pairs_array[:, 1]].resids   # problem line
```

**Comment by IAlibay** [comment]
If I'm reading this right, I'd suggest moving to using `atoms.ix` internally, which is 0-indexed (and is what everything internally should use for MDA).

**Comment by lilyminium** [comment]
I'd warn against using anything `ids`, `resids`, etc. because those are actually [labels that can be anything](https://github.com/MDAnalysis/mdanalysis/issues/3359#issuecomment-882521692) instead of real indices.

**Comment by orioncohen** [comment]
Thanks for tuning in! 

I didn't realize that ids and resids don't even have to be numeric. I use them in a couple places but clearly ix are a better choice. 

I'll make a PR switching to ix across the board.

**Comment by orioncohen** [comment]
Fixed by PR #39
